extensions/ipynb/src/notebookSerializer.ts

```diff
@@ -70,33 +91,67 @@ export class NotebookSerializer implements vscode.NotebookSerializer {
 		return data;
 	}
 
-	public serializeNotebook(data: vscode.NotebookData, _token: vscode.CancellationToken): Uint8Array {
-		return new TextEncoder().encode(this.serializeNotebookToString(data));
-	}
-
-	public serializeNotebookToString(data: vscode.NotebookData): string {
-		const notebookContent = getNotebookMetadata(data);
-		// use the preferred language from document metadata or the first cell language as the notebook preferred cell language
-		const preferredCellLanguage = notebookContent.metadata?.language_info?.name ?? data.cells.find(cell => cell.kind === vscode.NotebookCellKind.Code)?.languageId;
+	public async serializeNotebook(data: vscode.NotebookData, _token: vscode.CancellationToken): Promise<Uint8Array> {
+		if (this.disposed) {
+			return new Uint8Array(0);
+		}
 
-		notebookContent.cells = data.cells
-			.map(cell => createJupyterCellFromNotebookCell(cell, preferredCellLanguage))
-			.map(pruneCell);
+		if (this.experimentalSave && !this.isBrowser) {
+			return this.serializeViaWorker2(data);
+		}
+		const serialized = serializeNotebookToString(data);
+		return new TextEncoder().encode(serialized);
+	}
 
-		const indentAmount = data.metadata && 'indentAmount' in data.metadata && typeof data.metadata.indentAmount === 'string' ?
-			data.metadata.indentAmount :
-			' ';
-		// ipynb always ends with a trailing new line (we add this so that SCMs do not show unnecessary changes, resulting from a missing trailing new line).
-		return JSON.stringify(sortObjectPropertiesRecursively(notebookContent), undefined, indentAmount) + '\n';
+	private async startWorker() {
+		if (this.disposed) {
+			throw new Error('Serializer disposed');
+		}
+		if (this.worker) {
+			return this.worker;
+		}
+		const { Worker } = await import('node:worker_threads');
+		const outputDir = getOutputDir(this.context);
+		this.worker = new Worker(vscode.Uri.joinPath(this.context.extensionUri, outputDir, 'notebookSerializerWorker.js').fsPath, {});
+		this.worker.on('exit', (exitCode) => {
+			if (!this.disposed) {
+				console.error(`IPynb Notebook Serializer Worker exited unexpectedly`, exitCode);
+			}
+			this.worker = undefined;
+		});
+		this.worker.on('message', (result: { data: Uint8Array; id: string }) => {
+			const task = this.tasks.get(result.id);
+			if (task) {
+				task.complete(result.data);
+				this.tasks.delete(result.id);
+			}
+		});
+		this.worker.on('error', (err) => {
+			if (!this.disposed) {
+				console.error(`IPynb Notebook Serializer Worker errored unexpectedly`, err);
+			}
+		});
+		return this.worker;
+	}
+	private async serializeViaWorker2(data: vscode.NotebookData): Promise<Uint8Array> {
```

Comment:
Nitpick, should this be renamed to `serializeViaWorker` instead of `serializeViaWorker2`?